CMakeLists.txt

```diff
@@ -1,86 +1,66 @@
-cmake_minimum_required(VERSION 3.15)
-
-# Set the project name to your project name, my project isn't very descriptive
-project(myproject CXX)
-include(cmake/StandardProjectSettings.cmake)
-include(cmake/PreventInSourceBuilds.cmake)
-
-# Link this 'library' to set the c++ standard / compile-time options requested
-add_library(project_options INTERFACE)
+cmake_minimum_required(VERSION 3.16)
+
+# uncomment to set a default CXX standard for the external tools like clang-tidy and cppcheck
+# and the targets that do not specify a standard.
+# If not set, the latest supported standard for your compiler is used
+# You can later set fine-grained standards for each target using `target_compile_features`
+# set(CMAKE_CXX_STANDARD 17)
+
+# Add project_options v0.12.1
+# https://github.com/cpp-best-practices/project_options
+include(FetchContent)
+FetchContent_Declare(_project_options URL https://github.com/cpp-best-practices/project_options/archive/refs/tags/v0.12.1.zip)
+FetchContent_MakeAvailable(_project_options)
+include(${_project_options_SOURCE_DIR}/Index.cmake)
+
+# uncomment to enable vcpkg:
+# # Setup vcpkg - should be called before defining project()
+# run_vcpkg()
+
+# Set the project name and language
+project(myproject LANGUAGES CXX)
+
+# Initialize project_options
+# uncomment the options to enable them:
+project_options(
+      ENABLE_CACHE
+      # WARNINGS_AS_ERRORS
+      ENABLE_CPPCHECK
+      ENABLE_CLANG_TIDY
+      ENABLE_CONAN
+      ENABLE_COVERAGE
+      # ENABLE_IPO
+      # ENABLE_INCLUDE_WHAT_YOU_USE
+      # ENABLE_PCH
+      # PCH_HEADERS
+      # ENABLE_DOXYGEN
+      # ENABLE_USER_LINKER
+      # ENABLE_BUILD_WITH_TIME_TRACE
+      # ENABLE_UNITY
+      # ENABLE_SANITIZER_ADDRESS
+      # ENABLE_SANITIZER_LEAK
+      # ENABLE_SANITIZER_UNDEFINED_BEHAVIOR
+      # ENABLE_SANITIZER_THREAD
+      # ENABLE_SANITIZER_MEMORY
+      # CONAN_OPTIONS
+)
+# project_options is defined inside cmakelib
 target_compile_features(project_options INTERFACE cxx_std_17)
```

Comment:
Could it be possible to pass `cxx_std_17` as a parameter to the `project_options` macro? That would make it so that there's no need to modify the `project_options` library interface after it has been created. It could also help with setting a default value for `CMAKE_CXX_STANDARD` in [`project_options/src/StandardProjectSettings.cmake`](https://github.com/cpp-best-practices/project_options/blob/d0db61776eb02128b8c65b16e509b6ffe82630ee/src/StandardProjectSettings.cmake#L49).

Reply:
Setting a default `CMAKE_CXX_STANDARD` is a good practice. If not set, `project_options` simply set it to the latest version. This is consistent with the behaviour of other programming languages which default to the latest standard instead of some old compatibility standard.